daemon/daemon.go

```diff
@@ -186,35 +185,8 @@ func (d *Daemon) UpdateManifests(spec update.Spec) (job.ID, error) {
 		return id, errors.New("no type in update spec")
 	}
 	switch s := spec.Spec.(type) {
-	case update.ReleaseSpec:
-		return d.queueJob(func(jobID job.ID, working *git.Checkout, logger log.Logger) (*history.CommitEventMetadata, error) {
-			rc := release.NewReleaseContext(d.Cluster, d.Manifests, d.Registry, working)
-			result, err := release.Release(rc, s, logger)
-			if err != nil {
-				return nil, err
-			}
-
-			var revision string
-			if s.Kind == update.ReleaseKindExecute {
-				commitMsg := spec.Cause.Message
-				if commitMsg == "" {
-					commitMsg = commitMessageFromReleaseSpec(&s)
-				}
-				if err := working.CommitAndPush(commitMsg, &git.Note{JobID: jobID, Spec: spec, Result: result}); err != nil {
-					return nil, err
-				}
-				revision, err = working.HeadRevision()
-				if err != nil {
-					return nil, err
-				}
-				defer d.askForSync()
-			}
-			return &history.CommitEventMetadata{
-				Revision: revision,
-				Spec:     &spec,
-				Result:   result,
-			}, nil
-		}), nil
+	case release.Changes:
```

Comment:
Ah, because this is an interface, which both `Automated` and `ReleaseSpec` implement. Cool.